File: 06 - Java/03-02-00-OperadoresIncrementoDecremento-ULP.java
public class ProgramaPrincipal {

    public static void main(String args[]){
        // Operadores Incremento
        var a = 0;
        // Operador post-incremento
        System.out.println("Antes de incrementar: " + a);
        a++;
        System.out.println("Despues de incrementar (a++): " + a);
        // Operador pre-incremento
        a = 5;
        System.out.println("Antes de incrementar: " + a);
        ++a;
        System.out.println("Despues de incrementar (++a): " + a);

        // Operadores Decremento
        // Post-decremento
        a = 10;
        System.out.println("Antes de decrementar: " + a);
        a--;
        System.out.println("Despues de decrementar (a--): " + a);
        // Pre-decremento
        a = 20;
        System.out.println("Antes de decrementar: " + a);
        --a;
        System.out.println("Despues de decrementar (--a): " + a);

    }
}

File: 02 - C/03-02-00-OperadoresIncrementoDecremento-ULP.c
#include <stdio.h>

int main() {
	int a, b, c;
	// Operadores de Incremento: Pre-incremento y Post-incremento
	a = 0;
	// Post-incremento
	a++;//Se incrementa la proxima vez que utilicemos esta variable
	printf("Nuevo valor de a: %d", a);
	// Pre-incremento
	++a;
	printf("\nNuevo valor de a: %d", a);
	// ejemplo
	a = 5;
	b = 2;
	c = ++a * b++;
	printf("\nValor a: %d", a);// No tenia pendiente ningun incremento
	printf("\nValor b: %d", b);// Si tenia pendiente un incremento
	printf("\nValor c: %d", c);
	
	// Operador decremento: Pre-decremento y Post-decremento
	a = 5;
	b = 2;
	c = --a * b--;
	printf("\nValor a: %d", a);
	printf("\nValor b: %d", b);
	printf("\nValor c: %d", c);
}

